obs-studio-client/source/controller.cpp

```diff
@@ -194,11 +245,20 @@ std::shared_ptr<ipc::client> Controller::host(std::string uri) {
 	std::string workingDirectory = serverWorkingPath.length() > 0 ? serverWorkingPath : get_working_directory();
 
 	// Test for existing process.
-	std::string pid_path = "server.pid";
+	std::string pid_path;
+
 #ifdef _WIN32
-	std::vector<char> tmp(MAX_PATH + 2);
-	TCHAR tmp_len = GetTempPath(MAX_PATH + 1, tmp.data());
-	pid_path = std::string(tmp.data(), tmp.data() + tmp_len) + "\\" + pid_path;
+	std::basic_string<TCHAR> tmp;
+	tmp.resize(MAX_PATH + 1);
+	DWORD tmp_len = GetTempPath(MAX_PATH + 1, &tmp[0]);
```

Comment:
See above notice about GetVolumeInformation()->lpMaximumComponentLength

Reply:
GetTempPath will not return something higher than MAX_PATH + 1 (about 261 according to the documentation). If our path is longer than that, I'm not sure what the API will do exactly so my assumption is it will always be that length or lower. If that's not good enough, then we need to choose a different API.

Reply:
Oh I see that name isn't name here, it's path. What a variable name...